.yamato/upm-ci.yml

```diff
@@ -219,3 +221,13 @@ commit_ci:
         - "/.*/"
   dependencies:
     - .yamato/upm-ci.yml#package_tests_minimal
+
+msbuild_solution_ci:
+  name: msbuild solution
+  agent:
+    type: Unity::VM
+    image: package-ci/win10:stable
+    flavor: b1.large
+  commands:
+    - nuget.exe restore src/Unity.Mathematics.sln
+    - msbuild src/Unity.Mathematics.sln
```

Comment:
I haven't tried, but can't you just do 
```
dotnet build src/Unity.Mathematics.sln
```
instead? It will do the restore for you. I'd suggest trying [Yamato Remote](https://internaldocs.unity.com/yamato_continuous_integration/cli/) if you want to iterate on an actual Bokken VM (seems best).

Reply:
This is what I get:

```
dale@dale-1S2M:/Users/dale/code/Unity.Mathematics [fix-solution]> dotnet build src/Unity.Mathematics.sln
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Nothing to do. None of the projects specified contain packages to restore.
/usr/local/share/dotnet/sdk/5.0.403/Microsoft.Common.CurrentVersion.targets(1217,5): error MSB3644: The reference assemblies for .NETFramework,Version=v4.7.1 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks [/Users/dale/code/Unity.Mathematics/src/Tests/Unity.Mathematics.Tests.csproj]
/usr/local/share/dotnet/sdk/5.0.403/Microsoft.Common.CurrentVersion.targets(1217,5): error MSB3644: The reference assemblies for .NETFramework,Version=v4.7.1 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks [/Users/dale/code/Unity.Mathematics/src/Unity.Mathematics/Unity.Mathematics.csproj]

Build FAILED.

/usr/local/share/dotnet/sdk/5.0.403/Microsoft.Common.CurrentVersion.targets(1217,5): error MSB3644: The reference assemblies for .NETFramework,Version=v4.7.1 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks [/Users/dale/code/Unity.Mathematics/src/Tests/Unity.Mathematics.Tests.csproj]
/usr/local/share/dotnet/sdk/5.0.403/Microsoft.Common.CurrentVersion.targets(1217,5): error MSB3644: The reference assemblies for .NETFramework,Version=v4.7.1 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks [/Users/dale/code/Unity.Mathematics/src/Unity.Mathematics/Unity.Mathematics.csproj]
    0 Warning(s)
    2 Error(s)

Time Elapsed 00:00:00.37
```

If instead I try restore, I get this:

```
dale@dale-1S2M:/Users/dale/code/Unity.Mathematics [fix-solution]> dotnet restore src/Unity.Mathematics.sln
  Determining projects to restore...
  Nothing to do. None of the projects specified contain packages to restore.
```

It seems like there is some sort of configuration that needs to be done that allows `dotnet` to restore that's different from how nuget works, because nuget works fine. Is it possible for me to just install nuget?

```
dale@dale-1S2M:/Users/dale/code/Unity.Mathematics [fix-solution]> nuget restore src/Unity.Mathematics.sln
MSBuild auto-detection: using msbuild version '15.0' from '/Library/Frameworks/Mono.framework/Versions/5.20.1/lib/mono/msbuild/15.0/bin'.
Restoring NuGet package NUnit.ConsoleRunner.3.9.0.
Restoring NuGet package NUnit.Extension.TeamCityEventListener.1.0.3.
Restoring NuGet package NUnit.Extension.NUnitV2ResultWriter.3.5.0.
Restoring NuGet package NUnit.Extension.NUnitProjectLoader.3.5.0.
Restoring NuGet package NUnit.3.8.1.
Restoring NuGet package NUnit.Extension.VSProjectLoader.3.8.0.
Restoring NuGet package NUnit.Extension.NUnitV2Driver.3.6.0.
Restoring NuGet package NUnit.Console.3.9.0.
Adding package 'NUnit.ConsoleRunner.3.9.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Adding package 'NUnit.Extension.NUnitProjectLoader.3.5.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Adding package 'NUnit.Extension.TeamCityEventListener.1.0.3' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Adding package 'NUnit.Extension.NUnitV2ResultWriter.3.5.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Adding package 'NUnit.3.8.1' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Adding package 'NUnit.Console.3.9.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Adding package 'NUnit.Extension.NUnitV2Driver.3.6.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Adding package 'NUnit.Extension.VSProjectLoader.3.8.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Added package 'NUnit.Console.3.9.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Added package 'NUnit.Extension.NUnitV2ResultWriter.3.5.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Added package 'NUnit.Extension.NUnitProjectLoader.3.5.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Added package 'NUnit.Extension.VSProjectLoader.3.8.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Added package 'NUnit.Extension.TeamCityEventListener.1.0.3' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Added package 'NUnit.Extension.NUnitV2Driver.3.6.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Added package 'NUnit.ConsoleRunner.3.9.0' to folder '/Users/dale/code/Unity.Mathematics/src/packages'
Added package 'NUnit.3.8.1' to folder '/Users/dale/code/Unity.Mathematics/src/packages'

NuGet Config files used:
    /Users/dale/.config/NuGet/NuGet.Config

Feeds used:
    /Users/dale/.nuget/packages/
    https://api.nuget.org/v3/index.json

Installed:
    8 package(s) to packages.config projects
```

Reply:
I spent some time on this and pushed two commits but I can't get it working either. My attempts with `msbuild` fails with 
```
  C:\build\output\Unity-Technologies\Unity.Mathematics\src\Tests\Tests\Shared\TestMath.gen.cs(4198,10): error CS0616: 'TestCompilerAttribute' is not an attribute class [C:\build\output\Unity-Technologies\Unity.Mathematics\src\Tests\Unity.Mathematics.Tests.csproj]
```
and it seems related to that Unity specific DLLs are not loaded.
Assuming that's something that is just taken care of when running the full IDE I tried
```
devenv /build Debug src/Unity.Mathematics.sln
```
but then it bails out saying it's not licensed.

I'd vote for going back to msbuild but read up on how the Unity DLLs are configured, so you can set that up from command line for the msbuild toolchain.